src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -299,6 +307,115 @@ static void AppendHashtableContents(IndentedTextWriter writer, Hashtable ht)
             }
         }
 
+        private static void EmitScan(IndentedTextWriter writer, RegexMethod rm, string id)
+        {
+            RegexOptions options = (RegexOptions)rm.Options;
+
+            // Emit locals initialization
+            //writer.WriteLine("this.quick = quick;");
+            //writer.WriteLine("base.runtextpos = textstart;");
+            //writer.WriteLine("base.runregex = regex;");
+            //writer.WriteLine("base.runtextstart = textstart;");
+            //writer.WriteLine("base.runtextbeg = textbeg;");
+            //writer.WriteLine("base.runtextend = textend;");
+            //writer.WriteLine("bool initialized = false;");
+            //writer.WriteLine();
+
+            //EmitTimeoutHandling();
+            //writer.WriteLine();
+
+            // Source generator doesn't support Right-To-Left so there is no need to add the sepcial bump logic.
+            // If Right-to-left is ever added to the source generator, then we would need to the logic to define
+            // bump, as well as stoppos
+            Debug.Assert((options & RegexOptions.RightToLeft) == 0);
+
+            EmitPrevLenCheck();
+            writer.WriteLine();
+
+            EmitMainSearchLoop();
+            writer.WriteLine();
+
+            return;
+
+#pragma warning disable CS8321 // Local function is declared but never used
+            void EmitTimeoutHandling()
+            {
+                writer.WriteLine("// Handle timeout argument");
+                writer.WriteLine("_timeout = -1;");
+                writer.WriteLine("bool ignoreTimeout = global::System.Text.RegularExpressions.Regex.InfiniteMatchTimeout == timeout;");
```

Comment:
How is timeout getting passed in?

Reply:
It now gets set during Run() before calling Scan, so no need to pass it into Scan any longer.